opencti-platform/opencti-front/src/utils/hooks/useOverviewLayoutCustomization.ts

```diff
@@ -1,9 +1,30 @@
+import { useEffect, useState } from 'react';
+import { useFragment } from 'react-relay';
+import { EntitySettingSettings_entitySetting$key } from '@components/settings/sub_types/entity_setting/__generated__/EntitySettingSettings_entitySetting.graphql';
+import { entitySettingFragment } from '@components/settings/sub_types/entity_setting/EntitySettingSettings';
 import useAuth from './useAuth';
 
-const useOverviewLayoutCustomization: (entityType: string) => Array<{ key: string, width: number }> = (entityType) => {
-  const { overviewLayoutCustomization } = useAuth();
-  return Array.from(overviewLayoutCustomization?.get(entityType)?.entries() ?? [])
-    .flatMap(([key, width]) => ({ key, width }));
+type OverviewWidgetLayout = { key: string, width: number, label: string };
+
+const useOverviewLayoutCustomization: (entityType: string) => OverviewWidgetLayout[] = (entityType) => {
+  const { entitySettings } = useAuth();
+  const entitySettingsData = entitySettings?.edges?.map((setting) => (
+    useFragment<EntitySettingSettings_entitySetting$key>(entitySettingFragment, setting.node)));
+
+  const [overviewLayoutCustomization, setOverviewLayoutCustomization] = useState<OverviewWidgetLayout[]>([]);
+
+  useEffect(() => {
+    const overviewLayoutCustomizationEntries = entitySettingsData
+      ?.map(({ target_type, overview_layout_customization }) => ({ key: target_type, values: overview_layout_customization }))
+      .filter((entry) => !!entry.values)
+      .map(({ key: entityTypeKey, values: widgetsValues }) => [entityTypeKey, widgetsValues]);
+    const overviewLayoutCustomizations = overviewLayoutCustomizationEntries
+      ? new Map(overviewLayoutCustomizationEntries.map(([key, values]) => [key, values]))
+      : new Map();
+    setOverviewLayoutCustomization(overviewLayoutCustomizations.get(entityType) ?? []);
+  }, [entitySettingsData, entityType]);
+
+  return overviewLayoutCustomization;
 };
```

Comment:
@JeremyCloarec I moved the code from useAuth/Root here after discussions.

I also removed the related unit tests ; this was too complex to setup due to fragments. 
I did not spend too much time on fixing them, because I believe we should have e2e tests instead.

Reply:
Shouldn't it be a useMemo ?

Reply:
Or even a useCallback on entityType ?

Reply:
definitely _not_ a useEffect indeed